mllib/src/main/scala/org/apache/spark/ml/optim/aggregator/LogisticAggregator.scala

```diff
@@ -253,14 +271,69 @@ private[ml] class LogisticAggregator(
     }
   }
 
+  /** Update gradient and loss using binary loss function. */
+  private def binaryUpdateInPlace(block: InstanceBlock): Unit = {
+    val size = block.size
+    val localGradientSumArray = gradientSumArray
+
+    // vec here represents margins or negative dotProducts
+    val vec = new DenseVector(Array.ofDim[Double](size))
+
+    if (fitIntercept) {
+      val intercept = coefficientsArray.last
+      var i = 0
+      while (i < size) {
```

Comment:
Would it be faster to fill an array with this value and then make a DenseVector? maybe I'm missing why not

Reply:
OK, I will update it.